lib/fluent/log.rb

```diff
@@ -53,9 +53,17 @@ def self.str_to_level(log_level_str)
       end
     end
 
-    def initialize(out=STDERR, level=LEVEL_TRACE, opts={})
-      @out = out
-      @level = level
+    def initialize(logger, opts={})
+      # overwrites logger.level= so that config reloading resets level of Fluentd::Log
+      orig_logger_level_setter = logger.class.public_instance_method(:level=).bind(logger)
+      me = self
+      # DaemonLogger's log level setting has difference of one level from that of this class.
+      # Then level in this class will be self.level + 1.
```

Comment:
Does DaemonLogger have same # of levels with Fluentd's? (It's just difference about origin 0 or 1?)
If so, good to note it.